corehq/apps/geospatial/utils.py

```diff
@@ -71,3 +71,33 @@ def get_lat_lon_from_dict(data, key):
     except (KeyError, BadValueError):
         lat, lon = ('', '')
     return lat, lon
+
+
+def features_to_points_list(features):
+    """
+    `features` should be in the following format:
+    {
+        feature_id: {
+            geometry: {
+                coordinates: [
+                    [
+                        [lon, lat]
+                        ...
+                    ],
+                    ...
+                ]
+            }
+        }
+    }
+    Each list in `coordinates` is a separate polygon of the feature. This will usually only be a single polygon,
+    but there may be more if a multi-polygon is given.
+    """
+    points_list = []
```

Comment:
I worry that this could cause problems. Combining the points of different features could result in incoherent shapes.

[GeoJSON](https://datatracker.ietf.org/doc/html/rfc7946) has [specific rules](https://datatracker.ietf.org/doc/html/rfc7946#section-3.1.5) for [how coordinates](https://datatracker.ietf.org/doc/html/rfc7946#section-3.1.3) must [be ordered](https://datatracker.ietf.org/doc/html/rfc7946#section-3.1.6).

I think that creating a `MultiPolygon` from multiple `Polygon`s would work with Elasticsearch. But I think that needs to be intentional, otherwise the shapes might not return the results that the user intends, or that the map represents.

Reply:
On second thoughts, I think leave this at it is, because it does what we need it to do here. We can follow up in the code that builds on this to cater for different geoshapes.

Reply:
I think given that the points from features get added and read sequentially in this list it should be fine. Additionally, we don't expect multi-polygons from the front-end. Even if the user draws or saves multiple shapes together they will still be separate polygons (unless we intentionally merge them together in the code).

I do agree this could be followed up with a refactor in the future. We'll also need to decide whether to combine the polygons together on the front-end, or handle it on the back-end.